framework/src/userobject/DomainUserObject.C

```diff
@@ -78,35 +76,46 @@ DomainUserObject::DomainUserObject(const InputParameters & parameters)
           continue;
 
         // these are blocks connected to our blocks
-        _interface_connected_blocks.insert(interface_connected_block);
+        _interface_connected_blocks[interface_bnd_id].insert(interface_connected_block);
       }
     }
   }
 }
 
 const MooseVariableFieldBase *
-DomainUserObject::getInterfaceFieldVar(const std::string & var_name, const unsigned int comp)
+DomainUserObject::getInterfaceFieldVar(const std::string & var_name,
+                                       const unsigned int comp,
+                                       const std::set<BoundaryID> * interfaces)
 {
   const auto * const field_var = getFieldVar(var_name, comp);
   mooseAssert(field_var, "We should not be able to return a null variable");
-  _interface_vars.insert(field_var);
+  if (interfaces)
+    _interface_var_blocks[field_var->number()] = *interfaces;
+  else
+    _interface_var_blocks[field_var->number()] = _interface_bnd_ids;
```

Comment:
I'd rather not store by number. This could be problematic if a user couples in both a nonlinear variable and an aux variable

Reply:
Oh, that makes me worry too. Do we have a magic treatment in MOOSE to give them different numbers?

Reply:
haha no but you can always rely on the pointers being different!

Reply:
Decided to just use name string. I think MOOSE made sure the variable names across all systems are unique.